bin/parse_kraken2report.py

```diff
@@ -0,0 +1,168 @@
+            to_remove.add(i)
+
+    return [item for idx, item in enumerate(listKrakenTaxonomicKeys) if idx not in to_remove]
+
+def generateDictForLookupOfTaxonomicSubentries(removedIncompleteTaxaList):
+    # This function generates a lookup dictionary that allows to get all subentries
+    # of a certain taxonomic entry (e.g. for 'Homo' it looks as follows:
+    # {'Homo':'Homo sapiens'})
+    lookupDict = {}
+
+    for entry in removedIncompleteTaxaList:
+        listOfTaxonomicNames = entry.split('\t')
+
+        while listOfTaxonomicNames:
+            newTaxomomicName = listOfTaxonomicNames.pop(0)
+
+            if newTaxomomicName in lookupDict:
+                lookupDict[newTaxomomicName].update(listOfTaxonomicNames)
+            else:
+                lookupDict[newTaxomomicName] = set(listOfTaxonomicNames)
+
+    for key in lookupDict:
+        lookupDict[key] = sorted(list(lookupDict[key]))
+
+    sortedKeys = sorted(lookupDict)
+    return {k: lookupDict[k] for k in sortedKeys}
+
+def getKeysWithIDs(krakenHierarchicalTaxonomyDict, results=None):
+    # itterate over the hierarchical taxonomy dictionary and create
+    # a mapping from taxonomic name to taxonomic id.
+    # This is done recursively and results=None is the inital state.
+    if results is None:
+        results = {}
+
+    for k, v in krakenHierarchicalTaxonomyDict.items():
+        if isinstance(v, dict):
+            if 'id' in v:
+                results[k] = v['id']
+            results.update(getKeysWithIDs(v, results))
+    return results
+
+def main():
+
+    args = parse_args()
+
+    krakenList = read_in_kraken2report(args.input)
+    krakenHierarchicalTaxonomyDict = krakenTaxonomy2hierarchy(krakenList)
+    listKrakenTaxonomicKeys = getAllKeys(krakenHierarchicalTaxonomyDict)
+    resultRemovedIncompleteTaxa = removeIncompleteTaxa(listKrakenTaxonomicKeys)
+    resultLookUpDict = generateDictForLookupOfTaxonomicSubentries(resultRemovedIncompleteTaxa)
+    try:
+        # Try if the taxon to assess for/to filter is in the database (represented by the lookup dictionary)
+        # and add all subentries of the 'tax2filter' plus the 'tax2filter' to a single list
+        resultToFilter = resultLookUpDict[args.tax2filter] + [args.tax2filter]
+    except KeyError:
+        raise KeyError("The taxaomic group/taxon you want to check for/filter out is not in the kraken database. Use a database that includes the taxonomic group or taxon or change the tax2filter parameter to something that is in the database.")
+
+    resultLookUpOfIDsForTaxonomicNameDict = getKeysWithIDs(krakenHierarchicalTaxonomyDict)
+    resultTaxNameID = []
+    for entry in resultToFilter:
+        tax_name = entry
+        tax_id = str(resultLookUpOfIDsForTaxonomicNameDict[tax_name])
+        resultTaxNameID.append(tax_id)
+
+    with open('taxa_to_filter.txt', "w") as f:
+        for entry in resultTaxNameID:
+            f.write(entry+'\n')
+
+if __name__ == "__main__":
+    sys.exit(main())
```

Comment:
Is the `sys.exit(main())` necessary?  Wouldn't it be fine with just `main()`?

Reply:
This will output the exitcode of main(), so i think it is best-practice